AP-protocol.md

```diff
@@ -307,30 +307,42 @@ TODO
+    RemoveChannel(NodeId),
+    Crash,
+    ChangePdr(f32),
+    SentPacket(Packet),
+    DroppedPacket(Packet, f32),
+}
+```
 
-`AddSender(crossbeam::Sender, dst_id)`: This command provides a node with a new crossbeam Sender to send messages to node `dst_id`.
+### Simulation commands
 
-`AddReceiver(mpsc::Receiver, src_id)`: This command provides a node with a new crossbeam Receiver to receive messages from node `src_id`.
+Below are documented the commands that can be exchanged between the **SC and any type of node** (drone, client, server):
 
-`Spawn(id, code)`: This command adds a new drone to the network.
+- **SC -> Node (any)**
 
-`SetPacketDropRate(id, new_pdr)`:
+  - `Crash`: This commands makes a node crash. Upon receiving this command, the node’s thread should return as soon as possible. The last thing the node should do is to confirm that it is terminating by sending a Crash command to the SC. 
+  - `AddChannel(NodeId, Sender<Packet>)`: This command tells a node to add a new **edge** to the node with id `NodeId` and to use the provided `Sender<Packet>` channel to communicate with it. Since we are assuming bidirectional communication, this command must be used **two times** (i.e. to add the edge from `A` to `B`, we must add the `Sender from A to B` and the `Sender from B to A`).
+  - `RemoveChannel(NodeId)`: This command tells a node to remove the edge to the node with id `NodeId`. Since we are assuming bidirectional communication, this command must be used **two times** (i.e. to remove the edge from `A` to `B`, we must remove the `Sender from A to B` and the `Sender from B to A`).
 
-### Simulation events
+- **Node (any) -> SC**
 
-The Simulation Controller can receive the following events from nodes:
+  - `Crash`: used by a node to confirm that it is terminating.
+  - `SentPacket(Packet)`: a node sends this command every time it sends a packet to another node (drone, client or server). In this way, the SC can keep track of the network activity.
 
-`Topology(node_id, list_of_connected_ids, metadata)`: This event indicates that node `node_id` has been added to the network and its current neighbors are `list_of_connected_ids`. It can carry metadata that could be useful to display, such as the PDR and DR of Drones.
+In addition to the commands above, the **SC** and the **drones** can **exclusively exchange** the following commands:
 
-`MessageSent(node_src, node_trg, metadata)`: This event indicates that node `node_src` has sent a message to `node_trg`. It can carry useful metadata that could be useful display, such as the kind of message, that would allow debugging what is going on in the network.
+- **SC -> Drone**
+  - `ChangePdr(f32)` this command tells a drone to change its Packet Drop Rate to the new value provided.
 
+- **Drone -> SC**
+  - `DroppedPacket(Packet, f32)`: the drone sends this command every time it drops a packet. The `f32` value is the *Packet Drop Rate* of the drone at the time the packet was dropped. 
```

Comment:
Not sure we need a PDR in this command

Reply:
I added it because yesterday I got suggested that it might be useful to display on the SC when a packet is dropped. Since it does not add any complication, I think it is fine to have it for those who want to use it

Reply:
Yeah but the SC knos the pdr already

Reply:
Right, I guess it depends on the group implementation but somewhere in their topology they should have the PDR (which at this point I think should be changed accordingly when using the Chandepdr command). Ill revert it back and close this review. 

PS: If you can check my other responses it would be fine, so I can review / close also those.